### ISSUE TITLE
Quill removes the entire word when using the space after a word with a link

### ISSUE THREAD

**Issue by stirelli** [quilljs/quill#2407]
Quill removes the entire word when using the space after a word with a link

**Steps for Reproduction**

1. Visit https://codepen.io/stirelli/pen/GwyWjB
2. Select the word 'links'
3. Set a link
4. Make an space at the end of the word

**Expected behavior**:
Should make a normal space

**Actual behavior**:
Quill removes the entire word

**Platforms**:
Chrome 70.0.3538.77 (Official Build) (64-bit)
Ubuntu 18.10

**Version**:
v2.0.0-dev.3

**Comment by jaredly** [comment]
Looks like it only happens when the text is bold.
Additional information:
- if you have **some [bold](#) text** with a link in the middle, putting a space after the link deletes the link & puts the cursor at the end of the bold text.

**Comment by krogersforbes** [comment]
I confirmed that this also happens when text is in italics or underlined, I believe it's related to having the <a> nested inside of another tag. I traced the actual removal to https://github.com/quilljs/parchment/blob/2.0/src/blot/abstract/parent.ts#L237 
inside of parchment, but I don't have enough depth of knowledge at the moment to understand why the problem is occurring. 

Checking out the mutations that cause it, it looks like there's quite a lot going on in there for adding a single character. That's pretty much where I'm getting stuck on tracing it though.

![image](https://user-images.githubusercontent.com/5313773/52234796-84cdeb00-2890-11e9-8f52-9e2dc9c82358.png)
